libs/common/src/change-stream/delete-document-with-audit-logs.ts

```diff
@@ -0,0 +1,66 @@
+import { ObjectId, TransactionOptions } from 'mongodb';
+import { add } from 'date-fns';
+import { AuditDetails, MongoDbCollectionName } from '../types';
+import { MongoDbClient } from '../mongo-db-client';
+import { generateAuditDatabaseRecordFromAuditDetails } from './generate-audit-database-record';
+import { InvalidEnvironmentVariableError } from '../errors';
+
+type DeleteDocumentWithAuditLogsParams = {
+  documentId: ObjectId;
+  collectionName: MongoDbCollectionName;
+  db: MongoDbClient;
+  auditDetails: AuditDetails;
+};
+
+export const deleteDocumentWithAuditLogs = async ({
+  documentId,
+  collectionName,
+  db,
+  auditDetails,
+}: DeleteDocumentWithAuditLogsParams) => {
+  const { DELETION_AUDIT_LOGS_DELETE_AFTER_SECONDS } = process.env;
+
+  if (!DELETION_AUDIT_LOGS_DELETE_AFTER_SECONDS) {
+    throw new InvalidEnvironmentVariableError('DELETION_AUDIT_LOGS_DELETE_AFTER_SECONDS not set');
+  }
+
+  const client = await db.getClient();
+  const session = client.startSession();
+
+  try {
+    // TODO: verify that these are the options we want
+    const transactionOptions: TransactionOptions = {
+      readConcern: { level: 'snapshot' },
+      writeConcern: { w: 'majority' },
+    };
+    // TODO: check the transaction works
+    await session.withTransaction(async () => {
```

Comment:
Need developer testing to verify the transaction works (I've tested myself but want a second opinion)

Reply:
(Might not be needed if I can write & trust the api tests)